api/cpp/platform.rs

```diff
@@ -92,6 +98,8 @@ struct CppPlatform {
     window_factory: unsafe extern "C" fn(PlatformUserData, *mut WindowAdapterRcOpaque),
     #[cfg(not(feature = "std"))]
     duration_since_start: unsafe extern "C" fn(PlatformUserData) -> u64,
+    set_clipboard_text: unsafe extern "C" fn(PlatformUserData, *const u8, usize, _clipboard: u8),
+    clipboard_text: unsafe extern "C" fn(PlatformUserData, _clipboard: u8) -> StringView,
```

Comment:
Normally, this should work:

```suggestion
    set_clipboard_text: unsafe extern "C" fn(PlatformUserData, &SharedString, _clipboard: u8),
    clipboard_text: unsafe extern "C" fn(PlatformUserData, &mut SharedString, _clipboard: u8) ,
```

You can pass SharedString as reference from Rust to C++ and vice-versa.  this will map to a `SharedString*` and `const SharedString*`  in C++

Reply:
I added this, however I wanted to avoid a reference out variable in the C++ api, so I had the Platform return its own SharedString and then that handle gets copied over the empty rust-owned shared handle. Let me know if that looks safe to you. also see my comment about `clipboard_text`